ESSArch_Core/auth/tests/test_permissions.py

```diff
@@ -0,0 +1,353 @@
+            'add_user',
+            'change_user',
+            'delete_user',
+        ]
+        self.user.user_permissions.add(*Permission.objects.filter(codename__in=perm_list))
+
+        # needs no permission
+        self.validate_get_list_and_get_retrieve(self.user)
+
+        # needs add permission
+        response = self.get_response('post', 'create', self.user)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data, "hello from create")
+
+        # needs change permission
+        response = self.get_response('put', 'update', self.user)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(response.data, "hello from update")
+
+        # needs change permission
+        response = self.get_response('patch', 'partial_update', self.user)
+        self.assertEqual(response.status_code, status.HTTP_206_PARTIAL_CONTENT)
+        self.assertEqual(response.data, "hello from partial_update")
+
+        # needs delete permission
+        response = self.get_response('delete', 'destroy', self.user)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(response.data, "hello from destroy")
+
+    def test_when_model_permissions_are_ignored_and_user_has_no_permissions(self):
+        class IgnoredModelPermissionView(TestView):
+            _ignore_model_permissions = True
+
+        def get_response(method_name, action_name, user, authenticated=True):
+            request = getattr(self.factory, method_name)('/')
+
+            if authenticated:
+                force_authenticate(request, user=user)
+
+            return IgnoredModelPermissionView.as_view({method_name: action_name})(request)
+
+        # needs no permission
+        self.validate_get_list_and_get_retrieve(self.user)
+
+        # needs add permission
+        response = get_response('post', 'create', self.user)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data, "hello from create")
+
+        # needs change permission
+        response = get_response('put', 'update', self.user)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(response.data, "hello from update")
+
+        # needs change permission
+        response = get_response('patch', 'partial_update', self.user)
+        self.assertEqual(response.status_code, status.HTTP_206_PARTIAL_CONTENT)
+        self.assertEqual(response.data, "hello from partial_update")
+
+        # needs delete permission
+        response = get_response('delete', 'destroy', self.user)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(response.data, "hello from destroy")
+
+    def test_when_view_has_extra_function(self):
+        class ExtraFunctionView(TestView):
+            @action(detail=True, methods=['create'])
```

Comment:
The action decorator [expects](https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing) `methods` to be provided with a list of HTTP methods

```suggestion
            @action(detail=True, methods=['post'])
```